tests/testthat/test_new_ode_model.R

```diff
@@ -1,3 +1,37 @@
+test_that("Double absorption models work appropriately", {
+  mod <- new_ode_model(
+    code = "dAdt[1] = -KA1*A[1]; dAdt[2] = -KA2*A[2]; dAdt[3] = KA1*A[1] + KA2*A[2] -(CL/V)*A[3]; ",
+    pk_code = NULL,
+    parameters = c("CL", "V", "KA1", "KA2"),
+    obs = list(cmt = 1, scale = 1),
+    dose = list(
+      cmt = 1,
+      duplicate = c(2),
+      bioav = c(0.3, 0.6, 0.9)
+    ),
+    lagtime = c(0, 1.5),
+    cpp = T
+  )
+  reg <- new_regimen(amt = 100, n = 12, interval = 12, type = "oral")
+  res <- sim(
+    mod,
+    regimen = reg,
+    parameters = list(
+      KA1 = 1,
+      KA2 = 1,
+      CL = 1,
+      V = 10
+    ),
+    only_obs = F,
+    output_include = list(variables=TRUE),
+    t_obs = seq(0, 24, .1)
+  )
+  expect_equal(attr(mod, "dose")$duplicate, 2)
+  expect_equal(round(res[res$comp == 1 & res$t == 0.1, ]$y, 2), 27.15) # abs comp 1
+  expect_equal(res[res$comp == 2 & res$t == 1.2, ]$y, 0) # abs comp 2: should still be 0 due to lagtime
+  expect_equal(round(res[res$comp == 2 & res$t == 1.6, ]$y, 1), 54.3) # abs comp 2: should be twice the amount of t=0.1&comp==1
+})
+
```

Comment:
Should we also have an example where we need to rely on cmt_mapping to associate the dose type with the right cmt (i.e., a mix of oral and infusion dosing) and use a 2-cmt absorption model?

Reply:
Maybe we can also have a test where the bioavailability is supplied as an expression? e.g.: `"(dose_cmt[i] == 1) ? Fi : 1.0"`

Reply:
As discussed on Slack, we should not use that syntax anymore, and update the models that use it.